File: tp/61037 - Paz, Rodrigo Federico/tp6/backend/models/compras.py
from sqlmodel import SQLModel, Field
from typing import Optional

class Compra(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    usuario_id: int
    fecha: str
    direccion: str
    tarjeta: str
    subtotal: float
    iva: float
    envio: float
    total: float

class ItemCompra(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    compra_id: int = Field(foreign_key="compra.id")
    nombre: str
    cantidad: int
    precio_unitario: float

File: tp/61719 - Amoedo, Pablo Isaías/tp6/frontend/app/services/compras.ts
'use client';

import { api } from '@/lib/api';

export interface CompraResumen {
  compra_id: number;
  fecha: string;
  total: number;
  estado: string;
}

export interface DetalleCompra {
  producto_id: number;
  nombre: string;
  cantidad: number;
  precio_unitario: number;
  subtotal: number;
}

export interface CompraDetalle {
  compra_id: number;
  fecha: string;
  direccion: string;
  tarjeta: string;
  productos: DetalleCompra[];
  subtotal: number;
  iva: number;
  envio: number;
  total: number;
}

export async function obtenerHistorialCompras(): Promise<CompraResumen[]> {
  try {
    const response = await api.get('/compras');
    return response.data.compras || [];
  } catch (error: any) {
    console.error('Error al obtener historial:', error);
    if (error.response?.status === 401) {
      throw new Error('No autenticado. Por favor inicia sesión nuevamente.');
    }
    throw error;
  }
}

export async function obtenerDetalleCompra(compra_id: number): Promise<CompraDetalle> {
  try {
    const response = await api.get(`/compras/${compra_id}`);
    return response.data;
  } catch (error: any) {
    console.error('Error al obtener detalle:', error);
    if (error.response?.status === 401) {
      throw new Error('No autenticado. Por favor inicia sesión nuevamente.');
    }
    throw error;
  }
}
